src/spdx/parser/rdf/snippet_parser.py

```diff
@@ -0,0 +1,112 @@
+        license_info_in_snippet.append(
+            get_correctly_typed_value(logger, license_info_in_snippet_node,
+                                      lambda x: parse_license_expression(x, graph, doc_namespace)))
+    license_comment = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.licenseComments)
+    copyright_text = parse_literal_or_no_assertion_or_none(logger, graph, snippet_node, SPDX_NAMESPACE.copyrightText)
+    comment = parse_literal(logger, graph, snippet_node, RDFS.comment)
+    name = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.name)
+    attribution_texts = []
+    for (_, _, attribution_text_literal) in graph.triples((snippet_node, SPDX_NAMESPACE.attributionText, None)):
+        attribution_texts.append(attribution_text_literal.toPython())
+
+    raise_parsing_error_if_logger_has_messages(logger, "Snippet")
+    snippet = construct_or_raise_parsing_error(Snippet,
+                                               dict(spdx_id=spdx_id, file_spdx_id=file_spdx_id, byte_range=byte_range,
+                                                    line_range=line_range, license_concluded=license_concluded,
+                                                    license_info_in_snippet=license_info_in_snippet,
+                                                    license_comment=license_comment,
+                                                    copyright_text=copyright_text, comment=comment, name=name,
+                                                    attribution_texts=attribution_texts))
+    return snippet
+
+
+def set_range_or_log_error(
+    byte_range: Optional[Tuple[int, int]], line_range: Optional[Tuple[int, int]], logger: Logger,
+    parsed_range: Dict[str, Tuple[int, int]]) -> Tuple[Optional[Tuple[int, int]], Optional[Tuple[int, int]]]:
+    if not parsed_range:
+        return byte_range, line_range
+    if "ByteOffsetPointer" in parsed_range.keys() and not byte_range:
+        byte_range = parsed_range["ByteOffsetPointer"]
+    elif "ByteOffsetPointer" in parsed_range.keys() and byte_range:
+        logger.append("Multiple ByteOffsetPointer found.")
+    elif "LineCharPointer" in parsed_range.keys() and not line_range:
+        line_range = parsed_range["LineCharPointer"]
+    elif "LineCharPointer" in parsed_range.keys() and line_range:
+        logger.append("Multiple LineCharPointer found.")
+    return byte_range, line_range
+
+
+def parse_ranges(start_end_pointer: URIRef, graph: Graph) -> Dict[str, Tuple[int, int]]:
+    range_values = dict()
+    for pointer_property in [POINTER_NAMESPACE.startPointer, POINTER_NAMESPACE.endPointer]:
+        try:
+            pointer_node = graph.value(start_end_pointer, pointer_property, any=False)
+        except UniquenessError:
+            raise SPDXParsingError([f"Multiple values for {pointer_property.fragment}"])
+        if not pointer_node:
+            raise SPDXParsingError([f"Couldn't find pointer of type {pointer_property.fragment}."])
```

Comment:
This should probably be more specific, as it only means that it couldn't find a startPointer in this particular startEndPointer node, not the whole document. As the node has no Id, though, this can't be too specific, either... ;)

Reply:
The error message will be wrapped with a ParsingError from a Snippet, I am not sure, how to specify this in more detail. As you pointed out the node has no ID. The other error messages do also not tell exactly where the error comes from only which element couldn't be parsed.